signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/azure/AzureKeyVaultSigner.java

```diff
@@ -58,13 +66,14 @@ public class AzureKeyVaultSigner implements Signer {
         useDeprecatedSignatureAlgorithm
             ? SignatureAlgorithm.fromString("ECDSA256")
             : SignatureAlgorithm.ES256K;
-  }
-
-  @Override
-  public Signature sign(byte[] data) {
-    final AzureKeyVault vault;
+    final AzureConnectionFactory connFactory = new AzureConnectionFactory();
+    final AzureConnectionParameters connectionParameters =
+        AzureConnectionParameters.newBuilder()
+            .withServerHost(constructAzureKeyVaultUrl(config.getKeyVaultName()))
+            .build();
+    this.azureConnection = connFactory.getOrCreateConnection(connectionParameters);
     try {
-      vault =
+      this.vault =
```

Comment:
I would consider connecting to the vault outside of the constructor. If it fails can it result in the object being left in an inconsistent state?

Reply:
The main reason of this change is that we don't want to keep a Signer in memory that failed to create the Vault object basically because they won't have any use afaik. Also we want to make sure that this is created only once which agree that we could do without having this in the constructor.

Reply:
think it could either go in an init() method called after constructor, or in a factory before construction

Reply:
changed this. should be clearer now